source/versioned-api/versioned-api.rst

```diff
@@ -0,0 +1,273 @@
+===================
+Unified Test Format
+===================
+
+:Spec Title: Versioned API
+:Spec Version: 1.0.0
+:Author: Andreas Braun
+:Advisors: Jeff Yemin, A. Jesse Jiryu Davis, Patrick Freed, Oleg Pudeyev
+:Status: Draft
+:Type: Standards
+:Minimum Server Version: N/A
+:Last Modified: 2020-11-05
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+As MongoDB moves toward more frequent releases (a.k.a. continuous delivery), we
+want to enable users to take advantage of our rapidly released features, without
+exposing applications to incompatible server changes due to automatic server
+upgrades. A versioned API will help accomplish that goal.
+
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`__.
+
+This document tends to use "SHOULD" more frequently than other specifications,
+but mainly in the context of providing guidance on writing test files. This is
+discussed in more detail in `Design Rationale`_.
+
+
+Specification
+=============
+
+Background
+----------
+
+When applications interact with MongoDB, both the driver and the server
+participate in executing operations. Therefore, when determining application
+compatibility with MongoDB, both the driver and the server behavior must be
+taken into account.
+
+An application can specify the server API version when creating MongoClient.
+When this is done:
+
+- The client sends the specified API version to the server, causing the server
+  to behave in a manner compatible with that API version.
+- The driver will behave in a manner compatible with the driver's behavior as of
```

Comment:
I'm finding this line is a bit hard to parse, could it be reworded to be a bit more simple / straightforward?

Reply:
What about this?
> The driver will behave in a manner compatible with the driver's original behavior at the time the specified API version was released.

Reply:
Hm, I think it's still a bit vague, e.g. what does "compatible" or "original behavior" mean in this case? This requirement  _has_ to be vague though because there isn't anything special for drivers to do yet, as is pointed out below. I think the issue here is that the only requirement we can feasibly impose is that the driver is compatible with a _server_ using that API version, not with _other versions of the driver_ using that API version. For instance, version 5.0 of a driver configured with `serverApiVersion: 1` does not necessarily need to exhibit the same behavior of version 6.0 of a driver configured with `serverApiVersion: 1` (e.g if 6.0 removed the `distinct` helper or something), though both versions do need to be able to _function_ with a server configured with `serverApiVersion: 1`.

So what about something as simple as the following:

>The driver will behave in a manner compatible with a server configured with that API version, regardless of the server's actual release version.

It's pretty weak, but I think it's the only meaningful requirement we can impose for now until more requirements emerge.

Reply:
I support Patrick's suggested wording.

Reply:
Sounds good, I applied your suggested wording.